src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -0,0 +1,223 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    // Clone an attribute and resolve it. 
+    // This can be tricky since some read-only properties are set via the constructor. 
+    // This assumes that the property name matches the constructor argument name. 
+    internal class AttributeCloner<TAttribute>
+        where TAttribute : Attribute
+    {
+        private readonly TAttribute _source;
+        private readonly ConstructorInfo _bestCtor;
+        private readonly Func<IReadOnlyDictionary<string, object>, object>[] _bestCtorArgBuilder;
+        private readonly INameResolver _nameResolver;
+
+        // If no name resolver is specified, then any %% becomes an error. 
+        class EmptyNameResolver : INameResolver
```

Comment:
I was wondering why our fxcop/stylecop rules weren't giving you errors for this, then I realized that when I enabled that for all the SDK repos I missed the core SDK repo. I've fixed that - if you sync master and rebase, you should now see a bunch of new errors to fix :)